sodium/src/secbuf.rs

```diff
@@ -232,16 +232,48 @@ impl SecBuf {
     /// helper for writing data to our internal buffer
     pub fn write(&mut self, offset: usize, data: &[u8]) -> Result<(), SodiumError> {
         if offset + data.len() > self.len() {
-            return Err(SodiumError::new("bad write offset / length"));
+            return Err(SodiumError::new("SecBuf write overflow"));
         }
-
         unsafe {
             let mut b = self.write_lock();
             std::ptr::copy(data.as_ptr(), (**b).as_mut_ptr().add(offset), data.len());
         }
+        Ok(())
+    }
+
+    // -- Helpers -- //
+
+    /// Check if the buffer is empty i.e. [0,0,0,0,0,0,0,0]
+    pub fn is_empty(&mut self) -> bool {
+        let buf = self.read_lock();
+        for i in 0..buf.len() {
```

Comment:
If we're going to implement utility functions like this, I feel we should defer to the sodium utilities : https://download.libsodium.org/doc/helpers#testing-for-all-zeros

Reply:
Done in secbuf_util.rs